collect/collect.go

```diff
@@ -355,56 +375,64 @@ func (i *InMemCollector) collect() {
 		// other.
 		select {
 		case <-i.done:
+			span.End()
```

Comment:
Can we defer `span.End()` instead of manually closing in a number of the select clauses?